ebos/eclproblem.hh

```diff
@@ -1459,103 +1459,105 @@ public:
     const InitialFluidState boundaryFluidState(unsigned globalDofIdx, const int directionId) const
     {
         OPM_TIMEBLOCK_LOCAL(boundaryFluidState);
-        FaceDir::DirEnum dir = FaceDir::FromIntersectionIndex(directionId);
-        assert(bcindex_(dir)[globalDofIdx] > 0);
```

Comment:
Why remove the assert()? What are the requirements/range for this index?

Reply:
Thanks for noticing. I just removed it during debugging and forgot to add it back.

Reply:
Seems like adding the assert() triggered a test failure. I will investigate.

Reply:
Since we now ask for the boundaryState at every boundary face (also the ones where no boundary conditions are set) we can not use assert here. I instead added a explicit check and return to avoid accessing the bcprops for  invalid indices.